src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.ts

```diff
@@ -299,6 +343,26 @@ export class ProgrammingExerciseUpdateComponent implements OnInit {
         this.programmingExercise.title = undefined;
     }
 
+    /**
+     * Sets the attribute invalidRepositoryNamePattern to an updated RegExp that does not allow auxiliary repository names that are already used for this exercise and only allows
+     * "-" besides [A-z]
+     */
+    private setInvalidRepoNamePattern() {
+        let invalidRepoNames = '';
+        this.programmingExercise.auxiliaryRepositories?.forEach((auxiliaryRepository) => (invalidRepoNames += '|' + auxiliaryRepository.name));
+        this.invalidRepositoryNamePattern = new RegExp('^(?!(solution|exercise|tests' + invalidRepoNames + ')\\b)\\b(\\w|-)+$');
+    }
+
+    /**
+     * Sets the attribute invalidDirectoryNamePattern to an updated RegExp that does not allow directory names that are already used for other auxiliary repositories of this
+     * exercise "-" besides [A-z]
```

Comment:
And numbers

Reply:
Fixed in https://github.com/ls1intum/Artemis/pull/3457/commits/371252a2c564e627b43c924b6c5e289cb1bd65bb